clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -2960,6 +2969,12 @@ static void RenderFloatingPointOptions(const ToolChain &TC, const Driver &D,
   }
 
   for (const Arg *A : Args) {
+    auto CheckMathErrnoForVecLib =
+        llvm::make_scope_exit([&, MathErrnoBeforeArg = MathErrno] {
+          if (NoMathErrnoWasImpliedByVecLib && !MathErrnoBeforeArg && MathErrno)
+            ArgThatEnabledMathErrnoAfterVecLib = A;
+        });
```

Comment:
I don't think we need such complexity.  Given all the variables exist you can just clear `ArgThatEnabledMathErrnoAfterVecLib` within the `OPT_fveclib` case block and then have

```
if (NoMathErrnoWasImpliedByVecLib && !ArgThatEnabledMathErrnoAfterVecLib && MathErrno)
  ArgThatEnabledMathErrnoAfterVecLib = A
```
just before we claim `A` after the switch block.

Reply:
Maybe a little contrived, but the suggested check will give a _kinda_ incorrect warning in some cases:

Given this (somewhat) contrived example:
`-fveclib=ArmPL -fno-fast-math -fno-math-errno -ffp-model=strict`

The current check will warn about `-ffp-model=strict` which is the option that in the end turns on `-fmath-errno`, the suggested check will warn about `-fno-fast-math`, which is not really correct as there's `-fno-math-errno` after it.

Reply:
I see. Fair enough.